compiler/rustc_attr_parsing/src/attributes/must_use.rs

```diff
@@ -0,0 +1,27 @@
+use rustc_attr_data_structures::AttributeKind;
+use rustc_feature::{AttributeTemplate, template};
+use rustc_span::{Symbol, sym};
+
+use crate::attributes::{AttributeOrder, OnDuplicate, SingleAttributeParser};
+use crate::context::{AcceptContext, Stage};
+use crate::parser::ArgParser;
+
+pub(crate) struct MustUseParser;
+
+impl<S: Stage> SingleAttributeParser<S> for MustUseParser {
+    const PATH: &[Symbol] = &[sym::must_use];
+    const ATTRIBUTE_ORDER: AttributeOrder = AttributeOrder::KeepLast;
```

Comment:
I chose KeepLast because it matches the previous behaviour.
I'm not sure how to determine whether KeepFirst or KeepLast is better

Reply:
for now, match old behavior. I wish this was consistent but it's not. Whatever breaks the fewest tests is what I use atm.

Reply:
At the very least, this setting now also influences the warning. In other words, the warning cannot point you at the wrong thing anymore